heudiconv/convert.py

```diff
@@ -574,9 +574,11 @@ def save_converted_files(res, item_dicoms, bids_options, outtype, prefix, outnam
             # For multi-echo sequences, we have to specify the echo number in
             # the file name:
             if bids_file and is_multiecho:
-                # Get the EchoNumber from json file info.  If not present, it's echo-1
-                echo_number = fileinfo.get('EchoNumber', 1)
-
+                # Get the EchoNumber from json file info.  If not present, use EchoTime
+                if 'EchoNumber' in fileinfo.keys():
+                    echo_number = fileinfo['EchoNumber']
+                else:
```

Comment:
We should ensure echotime is available
```suggestion
                elif fileinfo.get('EchoTime'):
```
and just set `echo_number` to None if neither are defined.

Reply:
I was thinking that, since this line should only be reached if the data are multi-echo, it should raise an error if EchoTime is unavailable. What was happening before (in #347) was that the real cause of the error (missing EchoNumber field) was being obscured because the value was defaulting to 1. The error I was getting there was from copying files, which was hard to interpret.

That said, maybe defaulting to `None` will make the resulting error clearer, while defaulting to `1` made it confusing.

Reply:
Either way is fine to me - let's keep it as you have it. Could you merge with master - the CI problem should be fixed now